### PR TITLE
Add autofrom Option to Email Notify Plugin

### PR DESCRIPTION
**Pull request by andocromn**
### Motivation for changes:
want to have the ability have the email from field automatically generated with the system hostname and username compatible with linux/bsd mta  

### Detailed changes:
- additional import of "os" to retrieve user name
- add bool "autofrom" to schema
- conditional to decide whether to use auto generate from address

### Config usage if relevant (new plugin or updated schema):
```
    notify:
      task:
        title: Example
        via:
          - email:
              to: admin@server.net
              smtp_host: smtpout.server.net
              autofrom: yes
```

### CODE REVIEW

File: flexget/plugins/notifiers/email.py [hunk 1]
```diff
@@ -164,7 +166,10 @@ def notify(self, title, message, config):
 
         email = MIMEMultipart('alternative')
         email['To'] = ','.join(config['to'])
-        email['From'] = config['from']
+        if config['autofrom']:
+                email['From'] = os.getlogin() + '@' + socket.getfqdn()
```
Comment (liiight) [inline flexget/plugins/notifiers/email.py]: Weird indentation. Probably used tab instead of spaces

File: flexget/plugins/notifiers/email.py [hunk 2]
```diff
@@ -164,7 +166,10 @@ def notify(self, title, message, config):
...
+        else:
+                email['From'] = config['from']
```
Comment (liiight) [inline flexget/plugins/notifiers/email.py]: Same indentation issue

File: flexget/plugins/notifiers/email.py [hunk 3]
```diff
@@ -164,7 +166,10 @@ def notify(self, title, message, config):
 
         email = MIMEMultipart('alternative')
         email['To'] = ','.join(config['to'])
-        email['From'] = config['from']
+        if config['autofrom']:
+            email['From'] = os.getlogin() + '@' + socket.getfqdn()
```
Comment (cvium) [inline flexget/plugins/notifiers/email.py]: `os.getlogin()` is not cross-platform

**Comment by liiight** [comment]
Have you tested this on windows by any chance?

**Comment by andocromn** [comment]
yeah I did test on windows and linux

**Comment by liiight** [comment]
Cool. Care to update the wiki?

**Comment by andocromn** [comment]
wiki updated

**Pull request merged**